src/wp-includes/class-wp-block.php

```diff
@@ -193,6 +217,21 @@ public function __get( $name ) {
 	 */
 	public function render( $options = array() ) {
 		global $post;
+
+		/**
+		 * Allows render_block() or WP_Block::render() to be short-circuited, by
+		 * returning a non-null value.
+		 *
+		 * @since 5.1.0
+		 *
+		 * @param string|null $pre_render   The pre-rendered content. Default null.
+		 * @param array       $parsed_block The block being rendered.
+		 */
+		$pre_render = apply_filters( 'pre_render_block', null, $this->parsed_block );
```

Comment:
Is it expected to use the parsed block data with all the filters applied in the constructor? I have no idea what implications it has, but I clearly see it might work different.

Reply:
Fixed in b1baaad559.